plugins/storage/volume/storpool/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/StorPoolDownloadTemplateCommandWrapper.java

```diff
@@ -53,7 +54,7 @@ public CopyCmdAnswer execute(final StorPoolDownloadTemplateCommand cmd, final Li
         String dstPath = null;
         KVMStoragePool secondaryPool = null;
         DataTO src = cmd.getSourceTO();
-        DataTO dst = cmd.getDestinationTO();
+        TemplateObjectTO dst = (TemplateObjectTO) cmd.getDestinationTO();
```

Comment:
Maybe ask if the object is instance of TemplateObjectTO, otherwise fail and log it?